server/services/utils/pdfFormatter/pdfFormatter.js

```diff
@@ -32,76 +36,73 @@ function formatPdfData(
     taggingCompany,
     approvedVersionDetails,
   }
-  return valueOrPlaceholder(allData, placeholder, templateName)
+  return valueOrPlaceholder(selectPathsFrom(allData), placeholder, templateName)
 }
 
-function valueOrPlaceholder(allData, placeholder, templateName) {
+function valueOrPlaceholder(dataSelector, placeholder, templateName) {
   const dateNow = moment().format('DD/MM/YYYY')
-
-  return Object.entries(pdfData[templateName]).reduce(readValuesReducer(allData, placeholder), {
-    values: { CREATION_DATE: dateNow },
-    missing: {},
-  })
-}
-
-const readValuesReducer = (allData, placeholder) => (summary, [key, spec]) => {
-  const value = readEntry(allData, spec)
-
-  if (value) {
-    return mergeWithRight(summary, {
-      values: { [key]: value },
-    })
-  }
-
-  return mergeWithRight(summary, {
-    values: { [key]: spec.noPlaceholder ? '' : placeholder },
-    missing: {
-      [spec.group]: {
-        [spec.required]: {
-          [key]: spec.displayName,
-        },
-      },
+  const pdfDataEntries = Object.entries(pdfData[templateName])
+  return pdfDataEntries.reduce(
+    (summary, [key, spec]) => {
+      const value = spec.paths
+        .map(dataSelector)
+        .filter(R.identity)
+        .join(spec.separator)
+
+      const newObject = value
+        ? {
+            values: { [key]: value },
+          }
+        : {
+            values: { [key]: spec.noPlaceholder ? '' : placeholder },
+            missing: {
+              [spec.group]: {
+                [spec.required]: {
+                  [key]: spec.displayName,
+                },
+              },
+            },
+          }
+
+      return mergeWithRight(summary, newObject)
```

Comment:
is `mergeWithRight` equivalent to : `{...summary, ...newObject }` ?

Reply:
No. Its a deep merge.  Spreading the objects overwrites nested data whereas mergeWithRight adds to existing data.